state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -407,6 +433,10 @@ impl L1Fetcher {
                     }
 
                     last_block_number_processed = block_number;
+                    let after = Instant::now();
+                    let duration = after.duration_since(before);
+                    let mut guard = metrics.lock().await;
+                    guard.parsing.add(duration);
```

Comment:
This is a bit of a nit, but my personal preference has always been doing this instead:
```suggestion
                    metrics.lock().await.parsing.add(duration);
```
Setting the lock to a variable might give off the impression that we're using the value inside the lock for more than one thing.

Reply:
OK